packages/core/src/storage/default-proxy-storage.ts

```diff
@@ -1,114 +1,95 @@
 import type { MessageType, StorageThreadType } from '../memory/types';
 import { MastraStorage } from './base';
 import type { TABLE_NAMES } from './constants';
-import type { DefaultStorage, type LibSQLConfig } from './libsql';
-import type { EvalRow, StorageColumn, StorageGetMessagesArg } from './types';
+import { InMemoryStorage } from './in-memory-storage';
+import type { EvalRow, LibSQLConfig, StorageColumn, StorageGetMessagesArg } from './types';
 
 /**
  * A proxy for the DefaultStorage (LibSQLStore) to allow for dynamically loading the storage in a constructor
+ * If the storage is in-memory, it will use the InMemoryStorage.
  */
 export class DefaultProxyStorage extends MastraStorage {
-  private storage: DefaultStorage | null = null;
-  private storageConfig: LibSQLConfig;
-  private isInitializingPromise: Promise<void> | null = null;
+  private storage: Promise<MastraStorage>;
 
-  constructor({ config }: { config: LibSQLConfig }) {
+  constructor({ config }: { config?: LibSQLConfig }) {
     super({ name: 'DefaultStorage' });
-    this.storageConfig = config;
-  }
-
-  private setupStorage() {
-    if (!this.isInitializingPromise) {
-      this.isInitializingPromise = new Promise((resolve, reject) => {
-        import('./libsql')
-          .then(({ DefaultStorage }) => {
-            this.storage = new DefaultStorage({ config: this.storageConfig });
-            resolve();
-          })
-          .catch(reject);
+    const url = config?.url || process.env.MASTRA_DEFAULT_STORAGE_URL;
+    if (!url || url === ':memory:') {
+      this.storage = Promise.resolve(new InMemoryStorage());
+    } else {
+      this.storage = new Promise((resolve, reject) => {
+        try {
+          import(['./', 'libsql'].join('')) // avoid automatic bundling
```

Comment:
Sadly, this can quickly become a foot gun, where we ignore optimizing these files. If we ever introduce a side effect, our tooling will not capture it.